test/e2e/simple_addi.mlir

```diff
@@ -0,0 +1,128 @@
+// RUN: circt-opt --pass-pipeline='handshake.func(canonicalize,lower-handshake-to-firrtl)' %s | circt-opt --pass-pipeline='firrtl.circuit(lower-firrtl-types,lower-firrtl-to-rtl)' | circt-translate --emit-verilog | FileCheck %s
```

Comment:
This test was extremely satisfying to write, but I can imagine it is also fairly brittle as things change. Is it worth having this kind of E2E test? If there is a better way to write such a test, I'm all ears.

Reply:
The question also came up for me in the FIRRTL-to-LLHD pass. You could argue that since FIRRTL is a primary input dialect, and FIRRTL is a more or less stable language, it could make sense to ingest standard FIRRTL, map to the FIRRTL dialect, lower, and emit Verilog as another "standard" language. I could see this making the test more stable in some cases (e.g. internal changes to the FIRRTL dialect).

Including the handshake dialect in the chain seems like making the test more brittle to me in this case, since it's yet another CIRCT-internal dialect that probably keeps evolving quickly. Maybe starting at a `.fir` file as primary input and going to `.v` is a shorter end-to-end test?

Reply:
Then again, the Verilog output is probably *super verbose* compared to the intermediate lowered FIRRTL, so it might make sense to check that rather than Verilog. Otherwise any minor change in what additional stuff the Verilog printer emits would end up breaking this test as well 😢

Reply:
I am starting to thing any e2e test like this that bottoms out in System Verilog will be pretty brittle. Maybe we should just keep the System Verilog tests as isolated unit tests.

I also agree that having Handshake in the chain makes it less stable than starting from FIRRTL, but I do wonder how we can incorporate integration testing starting from Handshake. If I'm trying to do HLS, I am considering Handshake as my input dialect to CIRCT, and FIRRTL is really just an implementation detail of how I get from Handshake to RTL-level dialects and System Verilog.

Maybe we could keep this test, but bottom out at the RTL dialect instead of translating to System Verilog? That would at least cut out the noise specific to System Verilog and let the test focus on the fact that Handshake operations generate the correct data and control paths, etc. at the RTL level.

Reply:
> Maybe we could keep this test, but bottom out at the RTL dialect instead of translating to System Verilog? That would at least cut out the noise specific to System Verilog and let the test focus on the fact that Handshake operations generate the correct data and control paths, etc. at the RTL level.

Yeah that sounds like a reasonable compromise.

Reply:
I took a look at this, and honestly at the moment I think it is best to remove such an end-to-end test. Starting from the Handshake conversion and lowering through the FIRRTL dialect to RTL still has a lot of ops related to bridging the FIRRTL and RTL worlds, like this for every connect:

```
      ...
      %4 = firrtl.asPassive %1 : (!firrtl.flip<uint<1>>) -> !firrtl.uint<1>                                                                                                                                        
      %5 = firrtl.stdIntCast %4 : (!firrtl.uint<1>) -> i1                                                                                                                                                          
      %6 = firrtl.stdIntCast %arg0_valid : (!firrtl.uint<1>) -> i1                                                                                                                                                 
      rtl.connect %5, %6 : i1
      ...
```

Nothing wrong with that on its own, but it's not exactly the minimal end-to-end test I was imagining. Maybe as things evolve more, we can add such a test.